hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -434,6 +434,6 @@ protected Configuration getHadoopConf() {
 
   protected String getLatestDataInstantTime() {
     return dataMetaClient.getActiveTimeline().filterCompletedInstants().lastInstant()
-        .map(HoodieInstant::getTimestamp).orElse(SOLO_COMMIT_TIMESTAMP);
+        .map(HoodieInstant::getTimestamp).orElse(HoodieTableMetadataUtil.createIndexInitTimestamp(SOLO_COMMIT_TIMESTAMP, 0));
```

Comment:
not sure if this solves/gives us much. 
1: if we happened to initialize more than 1 MDT partition, the initialization time will be different. its 010 suffix for 1st and 011 for 2nd. 
2: this api is used only in logging. 

So, may not be worth fixing it. atleast for this (getLatestDataInstantTime).

Reply:
I'm scared by the hardcode `-0`, it is hard to maintain, at least we should fine a constant for it.

Reply:
Makes sense. Let's actually do `orElseThrow(() -> new IllegalStateException("No completed instant in the metadata timeline."))` instead of setting some initial timestamp.